### PR TITLE
[MIG][website_mass_mailing_name] Migration to v10

### PR DESCRIPTION
**Pull request by yajo**
- Relicense to LGPL.
- Fix all known issues.
- Enable tour only in demo instances.
- Fix Sass headers.
- Remove compiled css and maps.
- Update JS modules to new api.
- Update tour to new tours api.
- Update module structure to match latest template.

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can't try on runbot and Travis is red (maybe not because this module), but it's difficult to get a correct interpretation. Let's try to fix the branch first.

**Comment by yajo** [comment]
Seems the [lack of sass](https://travis-ci.org/OCA/social/jobs/250665214#L1000) again.

**Comment by chienandalu** [comment]
It still fails. Runbot:
```"LoadError: cannot load such file -- bootstrap-sass
 
 This error occured while compiling the bundle 'web.assets_frontend' containing:
     - /web/static/lib/bootstrap/less/variables.less
     - /web/static/lib/bootstrap/less/mixins/vendor-prefixes.less
     - /web/static/lib/bootstrap/less/mixins/buttons.less
     - /web/static/src/less/variables.less
     - /web/static/src/less/utils.less
     - /web_editor/static/src/less/web_editor.variables.less
     - /web/static/src/less/import_bootstrap.less
     - /web_editor/static/src/less/web_editor.common.less
     - /website/static/src/less/website.ui.less
     - /website/static/src/less/website.ui.components.less
     - /website/static/src/less/website.wrapwrap.less
     - /website/static/src/less/website.snippets.less
     - /website/static/src/less/website.zoomodoo.less
     - /website_mass_mailing_name/static/src/css/website_mass_mailing_name.sass"```

**Comment by yajo** [comment]
Ahm of course, this adds sass to travis, not to runbot. How do I do that?

**Comment by pedrobaeza** [comment]
We are using travis2docker on runbot, but that kind of packages is not translated. Is there any way to install by "command line"?

**Comment by yajo** [comment]
I'm getting [this](https://travis-ci.org/OCA/social/jobs/251900547#L220-L221):

```
$ gem install bootstrap-sass
Fetching: execjs-2.7.0.gem (100%)
Successfully installed execjs-2.7.0
Fetching: autoprefixer-rails-7.1.2.1.gem (100%)
ERROR:  Error installing bootstrap-sass:
	autoprefixer-rails requires Ruby version >= 2.0.
```

Maybe the Travis gurus @moylop260 or @lasley can help here...

**Comment by lasley** [comment]
We need to make this easier. There's actually three required lines. I'll make the suggestion to sample travis, but here there are:
``` yaml
before_install:
 - rvm install ruby --latest
 - gem install bootstrap-sass
 - gem install compass --pre
```

**Comment by moylop260** [comment]
Offtopic: Could you review the beta js lint https://travis-ci.org/OCA/social/jobs/250665212#L527-L533?

**Comment by lasley** [comment]
Interesting this one is new to me `[W7903(javascript-lint), ] Expected dot to be on same line as object. [Error/dot-location]`. That makes me sad, I always put the dot prefacing the method (on the lower line) when using multiple lines

**Comment by moylop260** [comment]
Do you recommend us disable it?

**Comment by lasley** [comment]
We should probably leave enabled - IMO if it's in a Lint engine & I don't like it, I just don't know enough of the intricacies of the language to understand why it's a rule. 

Basically I've just been doing it wrong in ignorance for numerous years, so it's just going to be one of those things that feels clunky for a while & cause me to gripe 😆

File: website_mass_mailing_name/tests/test_ui.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+from openerp.tests.common import HttpCase
+
+
+class UICase(HttpCase):
+    def test_ui(self):
+        """Test snippet behavior."""
+        self.phantom_js(
+            "/",
+            "openerp.Tour.run('mass_mailing_partner', 'test')",
```
Comment (moylop260) [inline website_mass_mailing_name/tests/test_ui.py]: `odoo.__DEBUG__.services['web_tour.tour'].run('mass_mailing_partner', 'test')` & `odoo.__DEBUG__.services['web_tour.tour'].tours.mass_mailing_partner.ready` for v10.0?

**Comment by moylop260** [comment]
@yajo I just updated the .travis.yml in the last commit 🤕 
@chienandalu Could you test it again from runbot, please?

I tested the newsletter from https://runbot.odoo-community.org/runbot/205/179

testing

File: website_mass_mailing_name/tests/test_ui.py (hunk 1 above)
Comment (yajo) [reply website_mass_mailing_name/tests/test_ui.py]: This is now included in the migration commit, thanks!

**Review by chienandalu — COMMENTED** [review]
Everything seem to work fine. There's a strange behavior though:
- I inserted a block and chose the newsletter I wanted to point to.
- Entered from an anonymous session to the website and subscribed. I get the acknowledge and the email is subscribed.
- I refresh the page and the subscription form is now blocked:
image
- This also happens if a logged in user subscribes with his system email address.
- If a logged in user subscribes any other email address the form doesn't block.

Seems also that some of your tests may be failing.

**Comment by yajo** [comment]
This strange behaviour you see @chienandalu is Odoo's default thing. Just try in a bare runbot and you will see that it happens the same, with the difference being that there's no name field.

Also I fixed the tests, the tour definition was plain wrong, the tour call was wrong too, and besides it turns out you cannot switch the user in the middle of new v10 tours, I had to split it in logged tour and public tour, and run both sequentially.

I also rebased to get rid of #181, so in short it should go ✔️ now

**Comment by pedrobaeza** [comment]
This is still red

**Comment by yajo** [comment]
@moylop260 it seems compass, sass, or bootstrap-sass is not correctly installed. Could you check please?

**Comment by moylop260** [comment]
It seems similar to https://github.com/OCA/website/pull/361#issuecomment-316182210
I think that you should use the exactly version of these packages when worked

**Comment by moylop260** [comment]
@Yajo 
If locally works could you send us the packages versions, please?
 - `ruby --version`
 - `gem list`

The issues related comments are about versions:
 - https://github.com/Compass/compass/issues/1513#issuecomment-32276852

I just updated to `ruby 2.0` like as recommendation

**Comment by moylop260** [comment]
@Yajo 
Now we have the error 
```bash
2017-07-19 00:42:29,361 14623 WARNING openerp_test odoo.addons.base.ir.ir_qweb.assetsbundle: NoMethodError: undefined method `activate' for nil:NilClass
```

https://travis-ci.org/OCA/social/jobs/255080735#L1080

But the fixes now talk about the implementation of code
https://github.com/asciidoctor/asciidoctor-pdf/commit/b6a0df9e06a9b5640e6c68634cb8a745f46e558d

Could you review it, please?

**Comment by lasley** [comment]
Shortcut for this added in https://github.com/OCA/maintainer-quality-tools/pull/465

**Comment by yajo** [comment]
Let's see if it goes 💚 now

**Comment by moylop260** [comment]
The error `odoo.tests.common: phantomjs: u"ReferenceError: Can't find variable: MutationObserver`  is reproduced from runbot and It seems a phantomjs version error:

 - screen shot 2017-07-20 at 14 52 54

@Yajo you faced all posible environment errors in a week 😆 

I will update your travis.yml to fix it

cc @gurneyalex FYI

**Comment by moylop260** [comment]
Ok, now is red because other matter...

Currently MQT don't use `--db-filter` parameter to run testing:
 - https://travis-ci.org/OCA/social/jobs/255836472#L1046

For phantomjs test we need just one database using the `--db-filter` parameter (because the UI change if 2 database are enabled to connect) so we can fix it from MQT:

 - screen shot 2017-07-20 at 18 18 52


In fact, the feature is working well!

 - [IMAGE ALT TEXT HERE](https://www.youtube.com/watch?v=XrJs-BUoCYo)

**Review by moylop260 — APPROVED** [review]
Dummy commit using https://github.com/OCA/maintainer-quality-tools/pull/468 worked fine!

Travis is green 🎉 
 - screen shot 2017-07-20 at 19 44 23

(The LINT build error is unrelated https://github.com/OCA/maintainer-quality-tools/issues/467)

**Pull request merged**